pkg/apis/core/v1alpha1/cluster_types.go

```diff
@@ -0,0 +1,159 @@
+}
+
+// ClusterList
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type ClusterList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+	Items []Cluster `json:"items" protobuf:"bytes,2,rep,name=items"`
+}
+
+// ClusterSpec defines how to find the cluster we're running
+// containers on.
+//
+// Tilt currently supports connecting to an existing Kubernetes
+// cluster or an existing Docker Daemon (for Docker Compose).
+type ClusterSpec struct {
+	// Connection spec for an existing cluster.
+	Connect *ClusterConnection `json:"connect,omitempty" protobuf:"bytes,1,opt,name=connect"`
+}
+
+// Connection spec for an existing cluster.
+type ClusterConnection struct {
+	// Defines connection to a Kubernetes cluster.
+	Kubernetes *KubernetesClusterConnection `json:"kubernetes,omitempty" protobuf:"bytes,1,opt,name=kubernetes"`
+
+	// Defines connection to a Docker daemon.
+	Docker *DockerClusterConnection `json:"docker,omitempty" protobuf:"bytes,2,opt,name=docker"`
+}
+
+type KubernetesClusterConnection struct {
+	// The name of the kubeconfig context to use.
+	//
+	// If not specified, will use the default context in the kubeconfig.
+	Context string `json:"context,omitempty" protobuf:"bytes,1,opt,name=context"`
+
+	// The default namespace to use.
+	//
+	// If not specified, will use the namespace in the kubeconfig.
+	Namespace string `json:"namespace,omitempty" protobuf:"bytes,2,opt,name=namespace"`
+}
+
+type DockerClusterConnection struct {
+	// The docker host to use.
+	//
+	// If not specified, will read the DOCKER_HOST env or use the default docker
+	// host.
+	Host string `json:"host,omitempty" protobuf:"bytes,1,opt,name=host"`
+}
+
+var _ resource.Object = &Cluster{}
+var _ resourcestrategy.Validater = &Cluster{}
+
+func (in *Cluster) GetObjectMeta() *metav1.ObjectMeta {
+	return &in.ObjectMeta
+}
+
+func (in *Cluster) NamespaceScoped() bool {
+	return false
+}
+
+func (in *Cluster) New() runtime.Object {
+	return &Cluster{}
+}
+
+func (in *Cluster) NewList() runtime.Object {
+	return &ClusterList{}
+}
+
+func (in *Cluster) GetGroupVersionResource() schema.GroupVersionResource {
+	return schema.GroupVersionResource{
+		Group:    "tilt.dev",
+		Version:  "v1alpha1",
+		Resource: "clusters",
+	}
+}
+
+func (in *Cluster) IsStorageVersion() bool {
+	return true
+}
+
+func (in *Cluster) Validate(ctx context.Context) field.ErrorList {
+	// TODO(user): Modify it, adding your API validation here.
+	return nil
+}
+
+var _ resource.ObjectList = &ClusterList{}
+
+func (in *ClusterList) GetListMeta() *metav1.ListMeta {
+	return &in.ListMeta
+}
+
+// ClusterStatus defines the observed state of Cluster
+type ClusterStatus struct {
+	// The preferred chip architecture of the cluster.
+	//
+	// On Kubernetes, this will correspond to the kubernetes.io/arch annotation.
+	//
+	// On Docker, this will be the Architecture of the Docker daemon.
```

Comment:
Do we need to normalize between these or do they use the same values? 🙏

Reply:
clarified that we're going to use GOARCH style!

Reply:
Since arch comes from nodes in the cluster, we're all good until someone makes a multi-node, multi-arch cluster. Maybe note that fact/assumption of uniform, "cluster"-wide arch?

Reply:
yep, didn't intend to suggest that this was the only arch, added a clarification